internal/rego/sigstore/sigstore.go

```diff
@@ -52,23 +54,36 @@ const (
 	ignoreRekorAttribute                 = "ignore_rekor"
 	publicKeyAttribute                   = "public_key"
 	rekorURLAttribute                    = "rekor_url"
+	rekorPublicKeyAttribute              = "rekor_public_key"
 )
 
 var ociImageReferenceParameter = types.Named("ref", types.S).Description("OCI image reference")
 
 var sigstoreOptsParameter = types.Named("opts",
+	// The definition below helps the compiler determine if the rego code is valid without having to
+	// execute it. This is very helpful for policy authors as it integrates nicely with IDEs, but
+	// getting this right is tricky. StaticProperties are always required, while DynamicProperties
+	// are restricted to a single type. The approach below is a compromise. Not all parameters are
+	// required, unless it's not a string which is currently just the `ignore_rekor` parameter. This
+	// makes adding new parameters not a breaking change, unless, of course, it's not of type
+	// string.
 	types.NewObject(
 		[]*types.StaticProperty{
-			{Key: certificateIdentityAttribute, Value: types.S},
-			{Key: certificateIdentityRegExpAttribute, Value: types.S},
-			{Key: certificateOIDCIssuerAttribute, Value: types.S},
-			{Key: certificateOIDCIssuerRegExpAttribute, Value: types.S},
 			{Key: ignoreRekorAttribute, Value: types.B},
-			{Key: publicKeyAttribute, Value: types.S},
-			{Key: rekorURLAttribute, Value: types.S},
 		},
-		nil,
-	)).Description("Sigstore verification options")
+		types.NewDynamicProperty(types.S, types.S),
+	)).Description(
+	fmt.Sprintf("Sigstore verification options. Dynamic string properties: `%s`.",
+		strings.Join([]string{
+			certificateIdentityAttribute,
+			certificateIdentityRegExpAttribute,
+			certificateOIDCIssuerAttribute,
+			certificateOIDCIssuerRegExpAttribute,
+			publicKeyAttribute,
+			rekorURLAttribute,
+			rekorPublicKeyAttribute,
```

Comment:
It's not obvious why we're changing the static to dynamic for these others. I assume the reasoning is good, but consider mentioning it in the commit message.

Reply:
I tried, unsuccessfully, to explain this in the large commit block above. :sob: 

Can you help me with the specifics of what is not clear?

Reply:
Maybe this... I made them dynamic properties so they are not all always required. Without this, something like this will cause a compile error:

```rego
ec.signstore.verify_image(ref, {"public_key": my_key, "ignore_rekor": false})
```

You need to provide all attributes, always:

```rego
ec.signstore.verify_image(ref, {
  "public_key": my_key,
 "ignore_rekor": false,
  "certificate_identity": "",
  "certificate_identity_regexp": "",
  "certificate_oidc_issuer" "",
  ...
})
```

With this change, the first form works.

Reply:
Ok, added an explanation to the commit message. I'll merge once CI passes.

Reply:
Sorry, I didn't notice the detailed explanation in the comments, just went looking for a commit message.. :sweat_smile:

Reply:
Commit message is perfect!